### DISCUSSION TITLE
What's the difference between Dialog and AlertDialog. Also Toast and Sonnet?

### DISCUSSION THREAD

**Discussion by syabro** [shadcn-ui/ui#4485] [Q&A] [answered]
They look the same to me

**Comment by itaim18** [comment, marked as answer]
1. One key difference between Dialog and AlertDialog is that AlertDialog interrupts the user and require a response. that is why its native behaviour would'nt allow the user to click outside the modal to close the alert, but Dialog could allow closing with no response. basically, AlertDialog behaves closer to alert.
2. Toast is basically rendering a succinct message that is displayed temporarily. it is a very basic component and works very much as easily expected. Sonner on the other hand, is an opinionated toast component, what it mainly means is that its behavior is very much defined, such as: stacking toasts, animation, hovering over multipule toasts also give you a cool animation effect. you could try looking at this [blog](https://emilkowal.ski/ui/building-a-toast-component) and see what "opinionated" means in this context.
Hope this answers your question :)

**Reply by syabro** [reply]
Thanks!
